pypesto/profile/profile.py

```diff
@@ -86,139 +88,37 @@ def create_next_guess(x, par_index, par_direction_, profile_options_,
+        task = ProfilerTask(
+            current_profile=current_profile,
+            problem=problem,
+            optimizer=optimizer,
+            options=profile_options,
+            create_next_guess=create_next_guess,
+            global_opt=global_opt,
+            i_par=i_par
+        )
+        tasks.append(task)
 
-            # compute the current profile
-            current_profile = walk_along_profile(
-                current_profile=current_profile,
-                problem=problem,
-                par_direction=par_direction,
-                optimizer=optimizer,
-                options=profile_options,
-                create_next_guess=create_next_guess,
-                global_opt=global_opt,
-                i_par=i_par)
+    # execute the tasks with Engine
+    ret = engine.execute(tasks)
 
-    return result
-
-
-def walk_along_profile(
-        current_profile: ProfilerResult,
-        problem: Problem,
-        par_direction: int,
-        optimizer: Optimizer,
-        options: ProfileOptions,
-        create_next_guess: Callable,
-        global_opt: float,
-        i_par: int
-) -> ProfilerResult:
-    """
-    This function computes half a profile, by walking ahead in positive
-    direction until some stopping criterion is fulfilled. A two-sided profile
-    is obtained by flipping the profile direction.
-
-    Parameters
-    ----------
-    current_profile:
-        The profile which should be computed
-    problem:
-        The problem to be solved.
-    par_direction:
-        Indicates profiling direction (+1, -1: ascending, descending)
-    optimizer:
-        The optimizer to be used along each profile.
-    global_opt:
-        log-posterior value of the global optimum
-    options:
-        Various options applied to the profile optimization.
-    create_next_guess:
-        Handle of the method which creates the next profile point proposal
-    i_par:
-        index for the current parameter
-
-    Returns
-    -------
-    current_profile:
-        The current profile, modified in-place.
-    """
-
-    # create variables which are needed during iteration
-    stop_profile = False
-
-    # while loop for profiling (will be exited by break command)
-    while True:
-        # get current position on the profile path
-        x_now = current_profile.x_path[:, -1]
-
-        # check if the next profile point needs to be computed
-        if par_direction == -1:
-            stop_profile = (x_now[i_par] <= problem.lb_full[[i_par]]) \
-                or (current_profile.ratio_path[-1] < options.ratio_min)
-
-        if par_direction == 1:
-            stop_profile = (x_now[i_par] >= problem.ub_full[[i_par]]) \
-                or (current_profile.ratio_path[-1] < options.ratio_min)
-
-        if stop_profile:
-            break
+    # fill in the ProfilerResults at the right index
+    for r in ret:
+        result.profile_result.list[-1][r['par_ind']] = r['prof']
```

Comment:
Are these abbreviations already used elsewhere in pyPESTO? If not, could rename some things to be a bit more readable, e.g.

```suggestion
    for indexed_profile in indexed_profiles:
        result.profile_result.list[-1][indexed_profile['index']] = \
            indexed_profile['profile']
```